core/server/utils/image-size.js

```diff
@@ -0,0 +1,193 @@
+
+            // CASE: `.ico` files might have multiple images and therefore multiple sizes.
+            // We return the largest size found (image-size default is the first size found)
+            if (dimensions.images) {
+                dimensions.width = _.maxBy(dimensions.images, function (w) {return w.width;}).width;
+                dimensions.height = _.maxBy(dimensions.images, function (h) {return h.height;}).height;
+            }
+
+            imageObject.width = dimensions.width;
+            imageObject.height = dimensions.height;
+
+            return Promise.resolve(imageObject);
+        } catch (err) {
+            return Promise.reject(new errors.InternalServerError({
+                code: 'IMAGE_SIZE',
+                err: err,
+                context: imagePath.href
+            }));
+        }
+    }).catch(function (err) {
+        if (err.statusCode === 404) {
+            return Promise.reject(new errors.NotFoundError({
+                message: 'Image not found.',
+                code: 'IMAGE_SIZE',
+                statusCode: err.statusCode,
+                context: err.url || imagePath.href || imagePath
+            }));
+        } else if (err.code === 'ETIMEDOUT') {
+            return Promise.reject(new errors.InternalServerError({
+                message: 'Request timed out.',
+                code: 'IMAGE_SIZE',
+                statusCode: err.statusCode,
+                context: err.url || imagePath.href || imagePath
+            }));
+        } else {
+            return Promise.reject(new errors.InternalServerError({
+                message: 'Unknown Request error.',
+                code: 'IMAGE_SIZE',
+                statusCode: err.statusCode,
+                context: err.url || imagePath.href || imagePath
+            }));
+        }
+    });
+};
+
+// Supported formats of https://github.com/image-size/image-size:
+// BMP, GIF, JPEG, PNG, PSD, TIFF, WebP, SVG, ICO
+// ***
+// Takes the url or filepath of the image and reads it form the local
+// file storage.
+// getImageSizeFromFilePath returns an Object like this
+// {
+//     height: 50,
+//     url: 'http://myblog.com/images/cat.jpg',
+//     width: 50
+// };
+// if the image is found and dimensions can be fetched, and rejects with error, if not.
+/**
+ * @description read image dimensions from local file storage
+ * @param {String} imagePath
+ * @returns {object} imageObject or error
+ */
+getImageSizeFromFilePath = function getImageSizeFromFilePath(imagePath) {
+    var imageObject = {},
+        dimensions = {};
+
+    imagePath = utils.url.urlFor('image', {image: imagePath}, true);
+    imageObject.url = imagePath;
+
+    // storage needs the path without `/content/images/` prefix
+    imagePath = imagePath.replace(new RegExp('^' + utils.url.urlJoin(utils.url.urlFor('home', true), utils.url.getSubdir(), '/', utils.url.STATIC_IMAGE_URL_PREFIX)), '');
+
+    return storage.getStorage()
+        .read({path: imagePath})
```

Comment:
Lot's of people are using e.g. s3 to storage images. Are we sure that the call via the s3 library is faster than calling the url? 🤔